pkg/kubectl/cmd/config/get_contexts.go

```diff
@@ -0,0 +1,161 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	clientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	"k8s.io/kubernetes/pkg/kubectl"
+
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+)
+
+type GetContextsOptions struct {
+	configAccess clientcmd.ConfigAccess
+	nameOnly     bool
+	showHeaders  bool
+	contextNames []string
+	out          io.Writer
+}
+
+const (
+	get_contexts_long = `Displays one or many contexts from the kubeconfig file.`
+
+	get_contexts_example = `# List all the contexts in your kubeconfig file
+kubectl config get-contexts
+
+# Describe one context in your kubeconfig file.
+kubectl config get-contexts my-context`
+)
+
+// NewCmdgetContexts creates a command object for the "get-contexts" action, which
+// retrieves one or more contexts from a kubeconfig.
+func NewCmdConfigGetContexts(out io.Writer, configAccess clientcmd.ConfigAccess) *cobra.Command {
+	options := &GetContextsOptions{configAccess: configAccess}
+
+	cmd := &cobra.Command{
+		Use:     "get-contexts [(-o|--output=)name)]",
+		Short:   "Describe one or many contexts",
+		Long:    get_contexts_long,
+		Example: get_contexts_example,
+		Run: func(cmd *cobra.Command, args []string) {
+			outputFormat := cmdutil.GetFlagString(cmd, "output")
+			outputTypes := map[string]bool{"": true, "json": false, "yaml": false, "wide": false, "name": true, "go-template": false, "go-template-file": false, "jsonpath": false, "jsonpath-file": false}
```

Comment:
I was asked to have the standard --output help string. But I can't support all the options as resource_printer needs a runtime object. So I wanted to differentiate between when a user gives an option that is in the help string (seemingly legal) and total garbage. In the case of a legal-but-not-supported case I am doing what "kubectl config view" does (reset to the default, and print a message).